core/runtime/src/main/java/io/quarkus/runtime/ApplicationLifecycleManager.java

```diff
@@ -159,36 +159,36 @@ public static void run(Application application, Class<? extends QuarkusApplicati
         (exitCodeHandler == null ? defaultExitCodeHandler : exitCodeHandler).accept(getExitCode()); //this may not be called if shutdown was initiated by a signal
     }
 
-    private static void registerHooks() {
+    private static void registerHooks(final Consumer<Integer> exitCodeHandler) {
         if (ImageInfo.inImageRuntimeCode() && System.getenv(DISABLE_SIGNAL_HANDLERS) == null) {
-            registerSignalHandlers();
+            registerSignalHandlers(exitCodeHandler);
         }
         final ShutdownHookThread shutdownHookThread = new ShutdownHookThread();
         Runtime.getRuntime().addShutdownHook(shutdownHookThread);
     }
 
-    private static void registerSignalHandlers() {
-        final SignalHandler handler = new SignalHandler() {
+    private static void registerSignalHandlers(final Consumer<Integer> exitCodeHandler) {
+        final SignalHandler exitHandler = new SignalHandler() {
             @Override
             public void handle(Signal signal) {
-                System.exit(signal.getNumber() + 0x80);
+                exitCodeHandler.accept(signal.getNumber() + 0x80);
             }
         };
-        final SignalHandler quitHandler = new SignalHandler() {
+        final SignalHandler diagnosticsHandler = new SignalHandler() {
             @Override
             public void handle(Signal signal) {
                 DiagnosticPrinter.printDiagnostics(System.out);
             }
         };
-        handleSignal("INT", handler);
-        handleSignal("TERM", handler);
+        handleSignal("INT", exitHandler);
+        handleSignal("TERM", exitHandler);
         // the HUP and QUIT signals are not defined for the Windows OpenJDK implementation:
         // https://hg.openjdk.java.net/jdk8u/jdk8u-dev/hotspot/file/7d5c800dae75/src/os/windows/vm/jvm_windows.cpp
-        if (OS.getCurrent() == OS.WINDOWS) {
-            handleSignal("BREAK", quitHandler);
+        if (OS.WINDOWS.isCurrent()) {
```

Comment:
It was there before but I'm not sure it's a good idea to use an util coming from GraalVM svm artifact for such a low level thing.

Reply:
Changed to use `System.getProperty("os.name")`. We need an utility to reuse that in various places